barretenberg/Earthfile

```diff
@@ -110,7 +110,9 @@ barretenberg-acir-tests-bb.js:
     # Run ecdsa_secp256r1_3x through bb.js on node to check 256k support.
     RUN BIN=../ts/dest/node/main.js FLOW=prove_then_verify ./run_acir_tests.sh ecdsa_secp256r1_3x
     # Run a single arbitrary test not involving recursion through bb.js for UltraHonk
-    RUN BIN=../ts/dest/node/main.js FLOW=prove_and_verify_ultra_honk ./run_acir_tests.sh 6_array
+    RUN BIN=../ts/dest/node/main.js FLOW=prove_and_verify_ultra_honk ./run_acir_tests.sh 6_array assert_statement
+    # Run the prove then verify flow for UltraHonk. This makes sure we have the same circuit for different witness inputs.
+    RUN BIN=../ts/dest/node/main.js FLOW=prove_then_verify_ultra_honk ./run_acir_tests.sh 6_array assert_statement
```

Comment:
I added a `prove_then_verify` flow. I do lean towards we should only use this flow for testing in general, but that is out of scope here.

Reply:
you mean we should exclusively use `prove_then_verify` and ditch the AND variant altogether?

Reply:
Yeah, unless we think there is a case where someone will be proving and verifying with the same `bb` instance at once.

Reply:
Yeah I think its only real value is for testing because it takes the vk serialization out of the equation. But for the sake of testing the real system I agree we should make sure we thorough tests of the THEN flow